### ISSUE TITLE
Ensure label definition for all fields

### ISSUE THREAD

**Issue by loicteixeira** [wagtail/wagtail#7114]
### Is your proposal related to a problem?

Django uses the model/form field attribute name when displaying it to the user, unless a label was provided.

```python
# No label. `alt_text` will be used as `Alt text` and will not be translated.
class ImageInsertionForm(forms.Form):
    alt_text = forms.CharField()

# Label defined. `Width` will be used and will be translated as needed.
class URLGeneratorForm(forms.Form):
    width = forms.IntegerField(label=_("Width"), min_value=0)
```

When the label is missing, no translation can be provided.

### Describe the solution you'd like

Enforce the need to specify a label.

- Needs to only affect Wagtail and not projects build with Wagtail, therefore a linter is probably more adapted than a Django check.
- Could the auto-label generation be disabled?

### Describe alternatives you've considered

Allow default attribute names to be translated

- Maybe too magical?
- Impede refactoring as changing an attribute name would also change the translation? Although it would already cause migrations and therefore should be handled with care.

### Next Steps

Work out a way to enforce the presence of a label.

Only then we can open a PR to fix the existing source strings.

**Comment by thibaudcolas** [comment]
Related: #6739. I looked briefly for a way to enforce having a label and found https://github.com/kalekseev/django-extra-checks as an example project that does this kind of enforcement.

**Comment by loicteixeira** [comment]
Yes, Django [checks](https://docs.djangoproject.com/en/3.2/topics/checks/) could be handy, be we have to make sure it only runs against Wagtail code and not against users codebase.

Otherwise, a flake8 plugin would work just fine, I would suggest trying to contribute to [flake8-django](https://github.com/rocioar/flake8-django) instead of creating yet another one (however it would mean adding a few rules from that plugin to our ignore list but I don't think it's an issue really).
